src/usecase/product/create/create.product.unit.spec.ts

```diff
@@ -0,0 +1,61 @@
+import CreateProductUseCase from "./create.product.usecase";
+
+const productACreateTestingInput = {
+  type: "a",
+  name: "my testing product name",
+  price: 20000,
+};
+
+const productBCreateTestingInput = {
+  type: "b",
+  name: "my testing product name",
+  price: 20000,
+};
+
+const MockRepository = () => {
+  return {
+    find: jest.fn(),
+    findAll: jest.fn(),
+    create: jest.fn(),
+    update: jest.fn(),
+  };
+};
+
+describe("Unit test create product use case", () => {
+  it("should create a product of type A", async () => {
+    const productRepository = MockRepository();
+    const productCreateUseCase = new CreateProductUseCase(productRepository);
+
+    const output = await productCreateUseCase.execute(productACreateTestingInput);
+
+    expect(output).toEqual({
+      id: expect.any(String),
+      name: "my testing product name",
+      price: 20000,
+    });
+  });
+
+  it("should create a product of type B", async () => {
+    const productRepository = MockRepository();
+    const productCreateUseCase = new CreateProductUseCase(productRepository);
+
+    const output = await productCreateUseCase.execute(productBCreateTestingInput);
+
+    expect(output).toEqual({
+      id: expect.any(String),
+      name: "my testing product name",
+      price: 40000,
```

Comment:
setei o valor que ele vai voltar mesmo do useCase, não sei se o ideal aqui seria mockar o create, porque esta logica está na factory, acho que o melhor mesmo é tirar esta logica dela, e colocar na criação/initializer, sempre que for um produto do tipo B CRIA com este valor já, mas assim ele da o resultado desejável por enquanto